crates/red_knot_python_semantic/src/ast_node_ref.rs

```diff
@@ -89,17 +85,30 @@ where
     }
 }
 
-impl<T> PartialEq for AstNodeRef<T> {
+impl<T> PartialEq for AstNodeRef<T>
+where
+    T: PartialEq,
+{
     fn eq(&self, other: &Self) -> bool {
-        self.node.eq(&other.node)
+        if self.parsed == other.parsed {
```

Comment:
What is the cost of this comparison? It doesn't seem like `ParsedModule` implements `Eq` -- does this comparison work because it implements `Deref` to `Parsed<ModModule>`? In that case it seems like it might be doing a deep compare of the entire AST? But I'm probably missing something.

Reply:
Oh oops, I was looking at my local copy without the latest changes from the coarse-grained structs PR.